---
title: "Novel genetic control of migratory diapause in Australian monarch butterflies"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  word_document:
    reference_docx: "paper-files/style_ref.docx"
  pdf_document:
    keep_tex: true
    fig_caption: true
    latex_engine: pdflatex
    fig_width: 7
    fig_height: 6
bibliography: "paper-files/citations.bib"
csl: "paper-files/vancouver.csl"
header-includes:
  - \usepackage[left]{lineno}
  - \linenumbers
  - \usepackage{setspace}\doublespacing
  - \usepackage{sectsty}\sectionfont{\fontsize{12}{12}\selectfont}
  - \usepackage{sectsty}\subsectionfont{\normalfont\itshape\fontsize{12}{12}\selectfont}
  - \usepackage[round]{natbib}
indent: true
sansfont: Times New Roman
fontsize: 12pt
always_allow_html: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(snpR); library(data.table); library(ggplot2); library(dplyr);
library(cowplot); library(openxlsx); library(nlme); library(gggenes);
library(Biostrings)

library(rnaturalearth)
library(ggplot2)
library(sf)
library(ggspatial)
library(ggimage)

```

```{r base_map, include=FALSE}
world <- ne_countries(scale = 'small', returnclass = 'sf')

polygon <- st_polygon(x = list(rbind(c(-0.0001, 90),
                                     c(0, 90),
                                     c(0, -90),
                                     c(-0.0001, -90),
                                     c(-0.0001, 90)))) |>
  st_sfc() |>
  st_set_crs(4326)

sf::sf_use_s2(FALSE)


world <- world |> st_difference(polygon)

world_robin <- st_transform(world, crs = '+proj=robin +lon_0=180 +x_0=0 +y_0=0 +ellps=WGS84 +datum=WGS84 +units=m +no_defs')

sf::sf_use_s2(TRUE)

# world <- ggplot() +
#   geom_sf(data = world_robin, fill= "azure2")  + #background color for land
#   coord_sf(xlim = c(-70e5, 100e5),
#            ylim = c(-60e5, 58e5)) +
#   theme_bw() +
#   annotation_scale(text_cex = 1.5) + 
#   theme(axis.text = element_text(size = 20),
#         axis.title = element_text(size = 24)) +
#   annotation_north_arrow(style = north_arrow_fancy_orienteering(text_size = 20), location = "bl", 
#                          pad_y = grid::unit(.05, "native"))
# 
# world
# ggsave("results/map.jpg", world, "jpg", width = 15, height = 11) # note, need to annotate this with ranges and arrows to produce final image

world <- ggplot() +
  geom_sf(data = world_robin, fill= "#dec079")  + #background color for land
  coord_sf(xlim = c(-70e5, 100e5),
           ylim = c(-60e5, 58e5)) +
  theme_bw() +
  annotation_scale(text_cex = 1.5, bar_cols = c("#eb612e", "#dec079"), text_family = "serif", text_col = "#393e42", pad_y = unit(0.75, "cm")) +
  theme(axis.text = element_text(size = 20, color = "#393e42"),
        axis.title = element_text(size = 24),
        panel.grid = element_line(color = "#393e42")) +
  annotation_north_arrow(style = north_arrow_fancy_orienteering(text_size = 20,
                                                                text_col = "#393e42",
                                                                text_family = "serif",
                                                                fill = c("#eb612e", "#dec079"),
                                                                line_col = "#393e42"),
                         location = "bl",
                         pad_y = grid::unit(.05, "native"))

world
ggsave("Figure1_base.jpg", world, "jpg", width = 15, height = 11)
```


<!-- chunks located here to ensure everything calls correctly -->
```{r, read_sample_meta_get_numbers, echo=FALSE,message=FALSE,include=FALSE,warning=FALSE}
sample.meta <- read.table("../data/sample_metadata.txt", header = T, stringsAsFactors = F)
old.data <- data.table::fread("../data/Aus_2018_metadata.txt", header = T)
old.data[,fam := as.numeric(gsub("_.+", "", ID))]

meta.micah <- data.table::fread("../data/2016_mon_meta.txt", header = TRUE)
```

```{r import_and_diversity, echo=FALSE, message=FALSE, warning=FALSE,include=FALSE}
#==========prep data==========
# read in genotypes and metadata
sample.meta <- read.table("../data/sample_metadata_v4.txt", header = T, stringsAsFactors = F)

# remove samples with no phenotypic data
# nas <- which(is.na(sample.meta$likely_pheno))
# genos <- genos[,-which(is.na(sample.meta$likely_pheno))]
# sample.meta <- sample.meta[-which(is.na(sample.meta$relaxed_pheno)),]

# add in colony info
fam_info <- sample.meta(readRDS("../colony/colony_R_outputs.RDS")$col_res$x)
fam_info$MotherID <- as.numeric(gsub("#", "", fam_info$MotherID))
fam_info$MotherID <- factor(fam_info$MotherID, levels = 1:max(fam_info$MotherID))
sample.meta$MotherID <- fam_info$MotherID[match(sample.meta$SampleID, fam_info$SampleID)] 
fam_info$FatherID <- as.numeric(gsub("\\*", "", fam_info$FatherID))
fam_info$FatherID <- factor(fam_info$FatherID, levels = 1:max(fam_info$FatherID))
sample.meta$FatherID <- fam_info$FatherID[match(sample.meta$SampleID, fam_info$SampleID)] 
sample.meta$cluster <- fam_info$ClusterIndex[match(sample.meta$SampleID, fam_info$SampleID)] 

# send year to factor, import, and filter
sample.meta$year <- as.factor(sample.meta$year)


#==========PCA plot==============
PCA <- fread("../scripts/association/PCAngsd_out.cov")
e <- eigen(as.matrix(PCA))
PCA <- e$vectors
PCA <- as.data.table(PCA)
colnames(PCA) <- paste0("PC", 1:ncol(PCA))
t.sizes <- c(18, 12)
FigureS1 <- ggplot(PCA, aes(x = PC1, y = PC2)) +
  geom_point(size = 3) +
  theme_bw() + 
  theme(legend.text = element_text(size = t.sizes[2]),
        axis.text = element_text(size = t.sizes[2]),
        axis.title = element_text(size = t.sizes[1]),
        axis.text.y = element_text(size = t.sizes[2]),
        legend.title = element_text(size = t.sizes[1])) +
  xlab(paste0("PC1 (", round(e$values[1]/sum(e$values)*100, 3), "%)")) +
  ylab(paste0("PC2 (", round(e$values[2]/sum(e$values)*100, 3), "%)"))

```

```{r Figure2, echo=FALSE, warning=FALSE, message=FALSE}
#==========family bias===================
bias <- sample.meta[,c("MotherID", "FatherID", "quant_pheno")]
bias$quant_pheno <- bias$quant_pheno/2
bias$MotherID <- as.factor(bias$MotherID)
bias$FatherID <- as.factor(bias$FatherID)

# permute to show expected random range per rank
perms <- matrix(0, 1000, length(unique(bias$MotherID)))
for(i in 1:nrow(perms)){
  tb <- bias
  tb$quant_pheno <- sample(tb$quant_pheno, nrow(tb), TRUE)
  perm_p <- tapply(tb$quant_pheno, tb$MotherID, mean)
  perm_p <- as.data.frame(perm_p)
  perm_p$MotherID <- 1:nrow(perm_p)
  colnames(perm_p)[1] <- "score"
  perm_p$rank <- rank(perm_p$score, ties.method = "first")
  perm_p$total <- as.numeric(table(tb$MotherID))
  perms[i,] <- perm_p$score[order(perm_p$rank)]
}
perm_summary <- data.table(score = colMeans(perms),
                           rank = 1:ncol(perms))
perm_summary[, c("lCI", "uCI") := as.data.frame(t(apply(perms, MARGIN = 2, FUN = function(x){
  return(t.test(x)$conf.int)
})))]

# anova to show significance
m1 <- lm(quant_pheno ~ MotherID, bias)
m1_an <- anova(m1)

# convert for plotting
bias_p <- tapply(bias$quant_pheno, bias$MotherID, mean)
bias_p <- as.data.frame(bias_p)
bias_p$MotherID <- 1:nrow(bias_p)
colnames(bias_p)[1] <- "score"
bias_p$rank <- rank(bias_p$score, ties.method = "first")
bias_p$total <- as.numeric(table(bias$MotherID))

# generate plot
t.sizes <- c(18, 12)
Figure2a <- ggplot(bias_p, aes(x = rank, y = score, size = total)) +
  geom_point(color ="#393e42") + theme_bw() +
  geom_line(data = perm_summary, aes(x = rank, y = score), inherit.aes = FALSE, color = "#eb612e") +
  geom_ribbon(data = perm_summary, aes(ymin = lCI, ymax = uCI, x = rank), 
              inherit.aes = FALSE, color = "#eb612e", alpha = .2) +
  xlab("Maternal Family") +
  ylab("Mean Reproductive Score") +
  labs(size = "Family Size")  +
  theme(legend.text = element_text(size = t.sizes[2] - 2),
        axis.text = element_text(size = t.sizes[2]),
        axis.title = element_text(size = t.sizes[1]),
        axis.text.y = element_text(size = t.sizes[2]),
        legend.title = element_text(size = t.sizes[2]),
        title = element_text(size = t.sizes[1] + 2)) +
  scale_size_continuous(range = c(5, 10)) +
  ggtitle("B")




# ggsave("~/2022-2023/ICCB2023/scatter.jpg", Figure3a, device = "jpg", dpi = 1000, height = 4, width = 8, scale = 3)


  

#==========association test -- read in from ANGSD============
chr_key <- fread("../data/chr_info.tsv")
chr_key$Chr_ID <- chr_key$`Chromosome name`
chr_key$Chr_ID <- make.unique(chr_key$Chr_ID)
chr_key$partID <- as.numeric(gsub(".+\\.", "", chr_key$Chr_ID))
chr_key$partID[-grep("\\.", chr_key$Chr_ID)] <- 0
chr_key <- dplyr::arrange(chr_key, as.numeric(`Chromosome name`), partID)
adat_quant <- data.table::fread("../scripts/association/association_out_quant_cov.lrt0.gz", header = T)
adat_quant$chr <- chr_key$Chr_ID[match(adat_quant$Chromosome, chr_key$`RefSeq seq accession`)]
adat_quant[grep("Un", chr), chr := "Other"]
adat_quant$pval <- pchisq(adat_quant$LRT, 1, lower.tail = FALSE)
adat_quant$lpval <- -log(adat_quant$pval)
adat_quant$fdr <- p.adjust(adat_quant$pval, "fdr")
adat_quant <- dplyr::arrange(adat_quant, pval)

q_quant <- plot_qq(adat_quant, "pval")
Figure2b <- ggplot(q_quant$.base$data, aes(x = .e, y = .o)) +
  geom_point(color = "#393e42") +
  geom_abline(slope = 1, intercept = 0, color = "#eb612e") +
  theme_bw() +
  theme(legend.text = element_text(size = t.sizes[2]),
        axis.text = element_text(size = t.sizes[2]),
        axis.title = element_text(size = t.sizes[1]),
        axis.text.y = element_text(size = t.sizes[2]),
        legend.title = element_text(size = t.sizes[1]),
        title = element_text(size = t.sizes[1] + 2)) +
  ylab(bquote(Observed ~ log[10](p))) +
  xlab(bquote(Expected ~ log[10](p))) +
  ggtitle("C")
  

# note -- color scheme for posters!
ord <- unique(chr_key$Chr_ID, sort = F)
ord <- ord[-grep("Un", ord)]
ord <- ord[-which(ord == "Z")]
ord <- c("Z", ord, "Other")
Figure2c <- plot_manhattan(adat_quant, "fdr", chr = "chr", bp = "Position",
                           significant = 0.05, suggestive = .1, log.p = TRUE,
                           chr_order = ord,
                           highlight = FALSE, colors = c("#eb612e", "#393e42"), 
                           t.sizes = c(t.sizes[1], t.sizes[1], t.sizes[2]), 
                           simplify_output = TRUE) # highlight top snp

# refetch chr centers, pulled from snpR
chr.info <- tapply(Figure2c$data[,"cum.bp"], Figure2c$data[,"chr"], max) # chromosome lengths
chr.centers <- rowMeans(cbind(tapply(Figure2c$data[,"cum.bp"], Figure2c$data[,"chr"], min), chr.info)) # chromosome centers

# plot
Figure2c <- Figure2c +
  ylab("-log10(p-value)") +
  xlab("Chromosome") +
  theme(legend.text = element_text(size = t.sizes[2]),
        axis.text = element_text(size = t.sizes[2]),
        axis.title = element_text(size = t.sizes[1]),
        axis.text.y = element_text(size = t.sizes[2]),
        legend.title = element_text(size = t.sizes[1]),
        title = element_text(size = t.sizes[1] + 2)) +
  scale_x_continuous(expand = c(0, 0), breaks = chr.centers, labels = 
                       names(chr.centers)) +
  ggtitle("D")

# ggsave("~/2022-2023/ICCB2023/manhattan.jpg", Figure3b, device = "jpg", dpi = 1000, height = 4, width = 8, scale = 3)

#==========egg photos (actual A)=========
imgs <- data.frame(category = c("Mature", "Partially Mature", "Immature"),
                   img = c("~/GitHub/aus_monarchs/paper/paper-files/mature.JPG",
                           "~/GitHub/aus_monarchs/paper/paper-files/partially_mature.JPG",
                           "~/GitHub/aus_monarchs/paper/paper-files/immature.JPG"),
                   x = 1,
                   y = 1)
imgs$category <- factor(imgs$category, levels = c("Mature", "Partially Mature", "Immature"))

eggplot <- ggplot(imgs, aes(x = x, y = y, image = img)) +
  geom_image(size = 1.3) +
  facet_wrap(~category, ncol = 1, strip.position = "left") +
  theme_minimal() +
  theme(strip.text = element_text(size = t.sizes[2], angle = 90),
        strip.placement = "outside",
        title = element_text(size = t.sizes[1] + 2),
        axis.text = element_blank(),
        axis.title = element_blank(),
        panel.grid = element_blank()) +
  ggtitle("A")


#==========combine=======================
f2al <- ggpubr::get_legend(Figure2a)
Figure2_top <- plot_grid(eggplot, Figure2a + guides(size = "none"), 
                         f2al, Figure2b, 
                         align = "hv", axis = "tlbr",
                         nrow = 1,
                         rel_widths = c(.5, 1, .2, 1)) # note, A is actually egg pictures, these labels are off

Figure2 <- plot_grid(Figure2_top,
                     Figure2c,
                     nrow = 2)

save_plot("Figure2.pdf", Figure2, base_height = 11, base_width = 15)

```

```{r re_analyze, include=FALSE}
old_phenotypic <- read.xlsx("../data/monarch_development.xlsx")

mod_AsRa <- lme(Aspect_Ratio ~ Yolked_Oocytes, random = ~ 1|Female_Line, old_phenotypic, na.action = na.omit)
mod_Area <- lme(Marea ~ Yolked_Oocytes, random = ~ 1|Female_Line, old_phenotypic, na.action = na.omit)
```

```{r genes_of_interest}
#=======identify targets=========
window <- 50*1000 # 10kb
targets <- adat_quant[fdr <= 0.05,]
targets[,start := Position - window]
targets[,end := Position + window]
targets <- dplyr::arrange(targets, Chromosome, start)

# one peak per chr, so simplify
targets[,start := min(start), by = chr]
targets[,end := max(end), by = chr]
targets <- unique(targets[,c("Chromosome", "chr", "start", "end")])

#=======locate genes=============
gtf <- rtracklayer::readGFF("../data/genomic.gtf")
gtf <- as.data.table(gtf)
gtf[,direction := ifelse(strand == "+", TRUE, FALSE)]
keep <- vector("list", nrow(targets))
all_genes <- keep
for(i in 1:nrow(targets)){
  keep[[i]] <- gtf[which(seqid == targets[i,Chromosome] &
                     ((end >= targets[i,start] & end <= targets[i,end]) |
                        (start <= targets[i,end] & start >= targets[i,start]) |
                        (start <= targets[i, start] & end >= targets[i, end]))),]
  all_genes[[i]] <- unique(keep[[i]][type == "gene",])
}
all_genes <- rbindlist(all_genes, idcol = "target")
IDs <- gsub("GeneID:", "", all_genes$db_xref)

# get the prot sequences from NCBI
cmd <- paste0("~/bin/datasets download gene gene-id ", paste0(IDs, collapse = " "))
system(cmd)

# copy the .faa and paste into monarchbase, read in the result here
prot_matches <- fread("../data/prot_matches_monarchbase.txt")
prot_matches[,LOC := unlist(stringi::stri_extract_all_regex(`Query ID`, "LOC.{9}"))]
prot_matches <- dplyr::arrange(prot_matches, LOC, Evalue)
prot_matches <- prot_matches[, head(.SD, 1), by= `Query ID`]
prot_matches <- dplyr::arrange(prot_matches, Length)
prot_matches <- prot_matches[, head(.SD, 1), by= LOC]

all_genes <- merge(all_genes, prot_matches, by.x = "gene_id", by.y = "LOC", all = TRUE)

#========plot====================
plot_parts <- vector("list", nrow(targets))
pmax <- max(-log10(adat_quant$fdr))


for(i in 1:nrow(targets)){
  # figure out max and min range
  min_pos <- targets[i,start]
  max_pos <- targets[i,end]
  
  tgtf <- keep[[i]]
  subgene <- tgtf[type == "CDS",]
  tgtf <- tgtf[type == "gene",]
  subgene[,gene_start := tgtf$start[match(gene, tgtf$gene)]]
  subgene[,gene_end := tgtf$end[match(gene, tgtf$gene)]]
  
  tgtf[,gene_lab := start + ((end - start)/2)]
  tgtf$OGS_name <- all_genes$`Monarch geneid`[match(tgtf$gene_id, all_genes$gene_id)]
  tgtf$OGS_name[is.na(tgtf$OGS_name)] <- tgtf$gene_id[is.na(tgtf$OGS_name)]

  subgene[,OGS_name := tgtf$OGS_name[match(gene, tgtf$gene)]]
  
  if(min_pos > min(c(tgtf$start, tgtf$end))){
    min_pos <- min(c(tgtf$start, tgtf$end))
  }
  if(max_pos < max(c(tgtf$start, tgtf$end))){
    max_pos <- max(c(tgtf$start, tgtf$end))
  }
  
  # plot SNPs
  
  tsnps <- adat_quant[which(Chromosome == targets[i,Chromosome] &
                     Position >= min_pos & Position <= max_pos),]
  
  pt <- ggplot(tsnps, aes(x = Position/1000000, y = -log10(fdr))) +
    geom_point() +
    theme_bw() +
    scale_x_continuous(limits = c(min_pos/1000000, max_pos/1000000)) +
    scale_y_continuous(expand = c(0, .05), limits = c(0, pmax))
  
  pmax <- max(c(pmax, max(-log10(tsnps$fdr))))
  
  if(i == 1)
    pt <- pt + 
    theme(axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.text.y = element_text(size = t.sizes[2]),
          axis.title.y = element_text(size = t.sizes[1]),
          axis.ticks.x = element_blank()) +
  ylab(bquote(-log[10](p[FDR])))
  
  if(i != 1){
    pt <- pt + 
      theme(axis.text.x = element_blank(),
          axis.title = element_blank(),
          axis.text.y = element_text(size = t.sizes[2]),
          axis.ticks.x = element_blank())
  }
  
  
  tgtf$OGS_name[tgtf$OGS_name == "DPOGS204613"] <- paste0("Karst/DPOGS204613")
  subgene$OGS_name[subgene$OGS_name == "DPOGS204613"] <- paste0("Karst/DPOGS204613")
  
  pb <- ggplot(tgtf, aes(xmin = start/1000000, xmax = end/1000000, 
                         y = OGS_name, forward = direction, fill = gene)) +
    geom_gene_arrow(arrow_body_height = unit(1, "lines"),
                    arrowhead_height = unit(1.5, "lines"), show.legend = FALSE,
                    fill = "#E5C16F") + 
    geom_subgene_arrow(data = subgene, 
                       aes(xmin = gene_start/1000000, xmax = gene_end/1000000, 
                           xsubmin = start/1000000, xsubmax = end/1000000,
                           y = OGS_name), 
                       fill = "black", arrow_body_height = unit(1, "lines"),
                       arrowhead_height = unit(1.5, "lines")) +
    theme_bw() +
    scale_x_continuous(limits = c(min_pos/1000000, max_pos/1000000)) +
    ylab("Gene") +
    xlab(paste0("Chromosome ", targets[i]$chr, " Position (mb)"))
  
  if(i == 1)
    pb <- pb + 
    theme(axis.text = element_text(size = t.sizes[2]),
          axis.title = element_text(size = t.sizes[1]))
  
  if(i != 1){
    pb <- pb + 
      theme(axis.title.y = element_blank(),
          axis.text = element_text(size = t.sizes[2]),
          axis.title.x = element_text(size = t.sizes[1]))
  }
  
  plot_parts[[i]] <- plot_grid(pt, pb, ncol = 1, rel_heights = c(1, 1),
                               align = "hv", axis = "tl")
  
}


Figure3 <- plot_grid(plot_parts[[1]],
                     plot_parts[[2]] + theme(axis.title.y = element_blank()),
                     plot_parts[[3]] + theme(axis.title.y = element_blank()), nrow = 1) +
  theme(plot.margin = unit(c(.1, .5, .1, .5), "cm"))

save_plot("Figure3.pdf", Figure3, base_height = 11, base_width = 15)

```


```{r mapping_stats}
sort_flagstat <- list.files("../data/v4/flagstat/", "\\.sort\\.flagstat", full.names = TRUE)
flt_flagstat <- list.files("../data/v4/flagstat/", "\\.sort\\.flt\\.flagstat", full.names = TRUE)

sort_flagstat <- unlist(lapply(sort_flagstat, readLines))
flt_flagstat <- unlist(lapply(flt_flagstat, readLines))

mapping <- data.table(sample = sample.meta$SampleID,
                      reads = as.numeric(gsub(" .+", "", sort_flagstat[seq(1, length(sort_flagstat), by = 16)])),
                      mapped = as.numeric(gsub(" .+", "", sort_flagstat[seq(7, length(sort_flagstat), by = 16)])),
                      retained = as.numeric(gsub(" .+", "", flt_flagstat[seq(1, length(sort_flagstat), by = 16)])))

total_reads <- sum(mapping$reads)
total_mapped <- sum(mapping$mapped)
total_retained <- sum(mapping$retained)

percent_mapped <- total_mapped/total_reads
percent_retained <- total_retained/total_reads

```

```{r Table_S1}

TS1 <- all_genes[,c("gene_id", "Monarch geneid", "seqid", "start", "end")]
TS1$Chromosome <- targets$chr[match(TS1$seqid, targets$Chromosome)]
colnames(TS1) <- c("RefSeq ID", "OGS ID", "Chromosome RefSeq ID", "Start", "End", "Chromosome")
TS1 <- TS1[,c("RefSeq ID", "OGS ID", "Chromosome", "Chromosome RefSeq ID", "Start", "End")]
TS1 <- dplyr::arrange(TS1, Chromosome, Start)
fwrite(TS1, "TableS1.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

```

# Abstract

Monarch butterflies (*Danaus plexippus*) are a charismatic and culturally important North American butterfly species famous for their dramatic and unique migratory life history. While non-migratory populations of the species are widespread and apparently stable, migratory populations in North America have recently seen declines, prompting concern that the migratory phenomenon in North America may be at risk of disappearing. In contrast, a relatively recently-established monarch population in Australia has rapidly re-acquired a migratory life history following hundreds of generations of residency and successive bottlenecks as the species island-hopped across the Pacific during the late 1800s and early 1900s. The process by which migration re-emerged in Australian monarchs is not currently known. We raised and sequenced individuals from Queensland, Australia under environmental conditions associated with migration initiation and found a novel association between reproductive diapause, a key migratory trait, and the spectrin beta chain protein *Karst*. This protein is known to be involved in reproductive development in monarchs but has not previously been identified to be associated with migratory life history variance. Our results provide an intriguing example of how the temporary loss of migration—in this case likely over hundreds of generations—may not entail the loss of genetic variation associated with this complex life history strategy.

# Introduction

Migratory species are acutely at risk from global anthropogenic environmental change [@Both2010; @chowdhurySeasonalSpatialDynamics2021]. The loss of breeding, wintering, or any transitional habitats may all cause declines in migrants [@Robbins1989], as may shifts in resource phenology due to seasonal changes which can cause resource availability and species life-histories to fall out of synchrony [@Jones2010a]. It is therefore not surprising that widespread declines have been observed in many migratory taxa [@Limburg2009;@Robbins1989;@rungeConservingMobileSpecies2014].

Population-level adaptability in migratory life-history should therefore be expected to increase long-term population viability by conferring resistance to phenological and environmental changes [@Pulido7341]. Migratory life-history adaptation can occur along three distinct axes: 1) *temporal*, or adaptation that changes the phenology of migration; 2) *spatial*, or adaptation that alters migratory pathways and orientation; and 3) *residency*, or adaptation that shifts populations between migratory and resident life histories. Inter-population variation and adaptation in each of these pathways is well known from nature: temporal changes or variation in migration are well documented in many taxa including as birds [@Rubolini2007; @Gordo2007; @chuCommunityScienceReveals2022; @prytulaRapidAdvancementSpring2022; @pinszkeLongtermChangesAutumn2023; @mondain-monvalFlywayscaleAnalysisReveals2021], fish [@kovachGeneticChangeEarlier2012], mammals [@shuertDecadalMigrationPhenology2022], and butterflies [@culbertsonLongtermMonitoringIndicates2022; @fitchettAdvanceTimingAnnual2022];  as is short-stopping, extension of migratory pathways, or complete re-direction of migration [@Ambrosini2011; @LaSorte2007; @Visser2009; @sonnleitnerRapidShiftsMigration2022;@rushingMigratoryBehaviorWinter2020;@dufourYellowbrowedWarblerPhylloscopus2022;@dufourNewWestwardMigration2021;@jiaShiftingMigrationRoute2021]; and shifts towards residency or partial migration have been observed and/or predicted in birds [@Pulido7341; @Plummer2015; @Møller2014;@haranPatternsPartialMigration2021; @falchiInterIntrapopulationVariability2023;@bonnetlebrunEffectsUrbanizationBird2020], butterflies [@Howard2010], mammals [@VanDerRee2006; @Fleming2003; @Jones2014a], and fish [@adamsOpinionPieceEvolutionary2022].

Adaptation via a transition to residency is of both particular interest and concern for the conservation of migratory species. While switching to a resident life history can allow species to avoid the increasing risks of migration and the intrinsically large energetic costs of migration in general [@McWilliams2004; @Rankin1992; @Wikelski2003], there is reasonable concern that the "extinction" of migration may not be easily reversible in some species. Permanent loss of migratory behavior is concerning because, in general, variation in migratory tendency is expected to increase long-term population persistence [@Pulido2007]. Beyond this, migratory species often provide critical ecosystem services [@lopez-hoffmanEcosystemServicesTransborder2017;@semmensAccountingEcosystemServices2011] and have cultural and/or ecological significance [@burgerComplexityBioindicatorSelection2015;@schuetzCharacterizingCulturalNiches2019;@semmensQuantifyingEcosystemService2018;@reynoldsMigratorySpeciesEcological2011] that may not survive the transition to permanent residency.

The loss of migratory behavior is a major concern for monarch butterflies (*Danaus plexippus*) [@browerAnimalMigrationsEndangered1991;@satterfieldMigratoryMonarchsThat2018]. Specifically, North American migratory monarch populations have experienced pronounced declines in counts of overwintering individuals over the last several decades, sparking a fear that those populations may go extinct in the near future [@Semmens2016; but see @crossleyOpposingGlobalChange2022]. However, many individual butterflies ostensibly from the North American migratory population are now year-round residents in some areas in the southern United States [@Howard2010], where they can survive due to the presence year-round of introduced tropical milkweed (*Asclepias curassavica*) [@bataldenPotentialChangesEastern2015], and there are widespread, non-migratory monarch populations that are not of any immediate conservation concern [@walkerDanausPlexippusErrata]. While the species as a whole may therefore not be at risk, their migratory behavior seemingly is. In addition to the damage to the long-term population persistence that could result from the loss of migration in North American populations (as discussed above), the loss of the publicly beloved annual monarch migration and concomitant overwintering would be culturally costly [@Gustafsson2015].

In contrast, there is a population of monarchs in Australia in which migratory behavior has apparently re-emerged after it was previously lost. This population, which was established sometime before 1871 [@Zalucki2004], is the product of a series of sequential introductions that crossed the Pacific Ocean from North America beginning sometime in the early 1840s after the introduction of their host plants enabled individuals to survive after being blown from the mainland [Figure 1, see @Zalucki2004; @Pierce2014; @Hemstrom2022a]. Monarchs are not known to be migratory on any Pacific islands, and no migratory behavior was observed in Australia until the 1930s, when non-reproductive overwintering aggregations were observed in northern New South Wales [NSW, see @James2019]. These and other individuals were later shown to migrate hundreds of kilometers each year to their wintering grounds from more southerly, inland locations in NSW [@James2019;@jamesInductionReproductiveDormancy1983]. Thus, more than 90 years (and hundreds of generations) after expanding out of North America, and more than 90 years without any evidence for seasonal migration, migration re-emerged in these populations. Understanding how this happened could provide crucial context around the "extinction" of migration in monarchs and other species.

To better understand how migration re-emerged in Australian monarchs, we studied continuously breeding monarch populations in Queensland. While these populations do not experience substantial seasonal fluctuations in resource availability, and thus do not migrate [@James2019], recent work has shown that reproductive diapause can be induced in some, *but not all* individuals from these populations if they are exposed to a reduction in photoperiod during development [@Freedman2018], a trait that these non-migratory populations do not display in the wild. In brief, reproductive diapause, where individuals delay their reproductive maturation, is a critical and fitness-relevant phenotype for migratory monarchs because it allows them to invest more heavily in lipid storage and increase their average longevity [@Herman2001;@denlingerInsectDiapauseRich2023;@numataDiapauseUnivoltineSemivoltine2023]. Like other migratory behaviors, diapause is initiated by changes in day length, temperature, and other seasonal fluctuations [@Goehring2002;@iiamsPhotoperiodicClockRegulation2019;@saundersDormancyDiapauseRole2020] and is thought to be directly controlled by juvenile hormone (JH) titers during development [@GreenII2019;@Herman2001;@denlingerInsectDiapauseRich2023]. 

While diapause is one of many migration-assoicated traits (such as wing morphology, resting and flight metabolic rates, lifespan, and lipid storage), it in particular is linked extremely tightly to migration. Nearly all wild migratory monarch females are in diapause: previous work has found that between 82 and 100% of female migratory monarch butterflies are non-reproductive [@majewskaExposureNonNativeTropical2019; @satterfieldMigratoryMonarchsThat2018a; @Goehring2002; @hermanStudiesAdultReproductive1981]. In contrast, previous rearing trials under typical North Americam summer breeding conditions in have yielded nearly 100% reproductively mature females [@Goehring2002]. These trends are less pronounced in males monarchs, which are therefore not the focus of this study [@majewskaExposureNonNativeTropical2019; @satterfieldMigratoryMonarchsThat2018a]. Given this, diapause serves as an excellent proxy for migration and has been used as such or as a focal trait for the study of migration in monarchs in many studies [see @satterfieldMigratoryMonarchsThat2018; @Goehring2002; @Tenger-Trolander2019; @GreenII2019; @majewskaExposureNonNativeTropical2019; @Freedman2018, for example]. 

The variance in diapause induction in Queensland monarchs allows for a direct search for genetic associations with this trait and thus for an examination of the genetic underpinnings of the re-evolution of migration in Australian monarchs. Here, we used reduced representation sequencing to search for genes associated with migratory diapause in Queensland monarch butterflies. We found three strongly associated loci, one of which co-located with spectrin beta chain protein involved in actin filament binding, a homolog of which is known to be expressed in silkworm ovaries. This gene had not previously been identified as associated with migration in North American monarch butterflies, and we did not find that any previously identified migratory genes were associated with diapause in our study. 

# Materials and Methods

## Background

Although the ancestral range of monarch butterflies is believed to be in North and Central America, they have fairly recently expanded into South America and the Caribbean, and very recently to several locations throughout the Atlantic and Pacific [@oplerFieldGuideWestern1999; @Zhan2014; @haegerMonarchsAtlanticOcean2015]. In North America, the species is well known for its unique, multi-generational migratory life history wherein individuals dispersing northward in the spring do so over three to four generations and then return to their wintering grounds in a single step [@malcolmSpringRecolonizationEastern1993; @cockrellTimeTemperatureLatitudinal1993].

In Southern Florida, the Caribbean, Central America, South America, and throughout most of their introduced range monarchs are non-migratory [@dockxDirectionalStabilizingSelection2007;@altizerPopulationsMonarchButterflies2010;@northMonarchButterflyBiology2004]. Residency in these populations appears to be a derived trait which has arisen multiple times after the species' split from *Danaus erippus*, their closest extant relative [@Zhan2014]. The Pacific expansion of monarchs is relatively recent, with historical records indicating that they established first in Hawaii in the 1840s, likely as a result of individuals blown in during storm events that were able to survive on recently introduced milkweeds [@clarkeMonarchsAustraliaWinds2004; @Pierce2014]. Genetic evidence is consistent with an introduction to Hawaii during this time period [@Hemstrom2022a]. As described above, they then spread across the Pacific and reached Australia in approximately 1871 [@Zalucki2004], and were recorded migrating again in Australia sometime in the 1930s [see Figure 1, @James2019].

## Sample collection, incubation, and phenotyping

We collected 22 female monarch butterflies from Pinjarra Hills, Queensland, Australia (27°32′26.7″S, 152°54′22.7″E) between the 5th and 9th of July 2018. Individuals in all life stages were present at the field site during our sample collections, consistent with continuous, year-round breeding. We enclosed each of these 22 females individually on either *Asclepias curassavica*  or *Gomphocarpus sp.* host plants found on-location and subsequently successfully collected eggs produced by 21 of them. While these eggs represent 21 total maternal families, each may consist of multiple groups of half-siblings since females may mate multiple times and lay clutches with mixed parentage. We did not remove *Ophryocystis elektroscirrha* (a protozoan parasite) spores, which were uniformly present on our eggs [consistent with previous reports from this location, see @Freedman2018].

We then incubated all eggs using the same incubators and following the "decreasing photoperiod" experimental scheme of Freedman et al. [-@Freedman2018] in order to induce reproductive diapause and ensure that our data could be later pooled with that produced in the earlier study. Briefly, larvae were reared at a constant temperature of 28°C under a photoperiod regime that declined from 14:10 L:D to 12:12 L:D over the course of 30 days (Δ4 mins/day). We released all male butterflies following emergence, since phenotyping diapause in male monarchs can be challenging due to the difficulty in extracting and weighing the ejaculatory duct, seminal vesicle, and accessory glands, which are likely the best indicator of male reproductive development [@Goehring2002]. This left us with a total of `r nrow(old.data)` females from `r length(unique(na.omit(old.data$fam)))` maternal families.

We assessed reproductive diapause by determining the degree of oocyte development in each of our adult female butterflies after 70 degree days of development, which is sufficient time for females to develop mature ovaries under normal summer conditions [@zaluckiEffectsAgeWeather1981]. We observed that individuals fell into several defined bins: `r sum(old.data$Phenotype == "C", na.rm = TRUE)` out of `r nrow(old.data)` had fully developed, chorionated eggs (visible ridges along the exterior of the egg), while `r sum(old.data$Phenotype != "C", na.rm = TRUE)` females did not. Of these latter individuals, a few (`r sum(old.data$Phenotype == "I", na.rm = TRUE) + sum(old.data$Phenotype == "N", na.rm = TRUE)`) had no yolk in their eggs whatsoever, but most had some degree of yolking. Since vitellogenesis generally only begins to a substantial degree after eclosion in monarchs [@Pan1976], we classified individuals as either fully reproductively mature (with chorionated eggs), partially reproductively mature (with more than a tiny amount of yolking), or reproductively immature (with no or only very small amounts of yolking). Images of egg development in each individual are available online (see Data Availability, Figure 2a). Wings and bodies from all individuals were preserved in dry coin envelops or 95% ethanol, respectively. We note that while we refer to the postponement of reproductive development under otherwise suitable conditions as diapause, other authors refer to this process in monarchs as oligopause [@jamesOvarianDormancyDanaus1982] or quiescence, since diapause implies an extended refractory period that we did not test for.

We supplemented these samples with preserved, dry butterfly wings from each of the "decreasing photoperiod" female individuals previously scored for reproductive development by Freedman et al. [-@Freedman2018], constituting an additional `r sum(sample.meta$year == 2016, na.rm = TRUE)` total female butterflies from `r length(unique(meta.micah$Female_Line[meta.micah$Treatment == "Decreasing"]))` maternal families, which were scored by the authors as reproductively chorionated, yolked, and unyolked, respectively. Together, these two datasets contained `r length(na.omit(sample.meta$quant_pheno))` monarch butterflies from `r length(unique(na.omit(sample.meta$MotherID)))` maternal families (see Table 1).


## Sequencing and genotyping
We removed and extracted DNA from one leg from each of our samples collected in 2018 or from a wing base from each of the samples collected by Freedman et al [-@Freedman2018] using the magnetic bead protocol of Ali et al. [-@Ali389]. We quantified the resulting DNA on a BioTek Instruments FLx800 Fluorescence Reader using Thermo Fisher Scientific Quant-iT PicoGreen dsDNA Reagent, then prepared Restriction Associated Digest (RAD) libraries using the Pst1 restriction enzyme according to Ali et al. [-@Ali389]. We sequenced these 150 bp paired-end sequencing libraries using an Illumina Hi-Seq 4000.

We aligned the resulting raw sequence data to the "MEX_DaPlex" monarch butterfly genome assembly [@ranzNovoTranscriptionalAtlas2021] using the mem algorithm of the Burrows-Wheeler Aligner [@Li2009a]. We then filtered out PCR duplicates, improperly paired, or poorly mapped reads using SAMtools [@Li2009]. For some downstream pedigree reconstruction, we then called genotypes using the ANGSD software [@Korneliussen2014] package with the following parameters: -doMajorMinor 1 (determine major and minor alleles using a genotype likelihood approach),  -doMaf 2 (determine minor allele frequencies), -SNP_pval 1e-8 (keep only loci with a SNP p-value $\le 1 \times 10^{-8}$), -doGeno 4 (call genotypes), -doPost 2 (calculate genotype posterior probabilities using a uniform prior), -postCutoff 0.95 (keep only loci where the highest genotype posterior probability $\ge 0.95$), -minQ 20 (keep only loci with a sequencing quality $\ge 20$), -minMapQ 20 (keep only loci with a mapping quality $\ge 20$), -minInd 130 (keep only loci sequenced in at least 130 individuals), and -minMaf 0.05 (keep only loci with a minor allele frequency $\ge 0.05$). 

<!-- we removed poorly seqeuenced individuals later in the pipeline, so these numbers are actually the final numbers. Not sure if that matters, but can fix. -->

## Statistical analysis
We first confirmed the parentage of our samples using the Colony2 program, assuming polygamous, random mating [@Jones2010]. In order to determine the degree to which diapause status was biased between families, we fit a basic linear model in R version 4.2.2 [@rcoreteamLanguageEnvironmentStatistical2022] using phenotypic status as the response variable and maternal family as a fixed effect. Phenotypic status was coded as 0, 1, or 2 for reproductively immature, partially mature, or mature, respectively. We used an ANOVA test in R to determine if maternal ID significantly improved model fit. To determine the expected phenotypic distribution for each family sorted by mean reproductive rank under a null distribution, we randomly permuted individuals between families 10,000 times and then ranked each family for reproductive score.

We conducted a Genome-Wide Association Study (GWAS) in order to determine if any loci were significantly associated with reproductive status. Since our data was generally of low coverage, we used a frequency test in ANGSD [@Korneliussen2014] based on genotype likelihoods rather than called genotypes and thus increase the effective number of loci we were able to analyze. We used the following parameters: -doMajorMinor 1, -SNP_pval 1e-12, -GL 1 (calculate genotype likelihoods using the SAMtools approach), -minQ 20, -minMapQ 20, -minInd 102 (50% of the individuals with called phenotypes), -minMaf 0.05, -doMaf 2 (assume a known minor allele), -doAsso 1 (do an association test using a frequency test), and -yquant coding phenotypic status as above. Since our samples were composed of many groups of full and half-siblings and came from two genetically different years, we also used the -cov argument and supplied the first 20 principal components derived from PCAngsd [@meisnerInferringPopulationStructure2018], which we ran using the default parameters using genotype likelihoods derived from ANGSD using the same parameters plus -doGlf 2 to save genotype likelihoods. We calculated p-values for the resulting likelihood ratio test score and corrected those values using the false discovery rate approach of Benjamini and Hochberg [-@benjaminiControllingFalseDiscovery1995a] in R. 

We constructed quantile-quantile and Manhattan plots of the resulting p-values using snpR [@hemstromSnpRUserFriendly2023] and identified candidate adaptive genes as those co-locating within 50kb of any SNPs with a corrected $p \le 0.05$. Since the "MEX DaPlex" reference genome uses RefSeq gene IDs but most monarch studies use the monarch official gene set IDs, we used BLAST [@Altschul1990] to identify official gene set genes with protein sequences matching those of the RefSeq genes we identified via the built-in tool on MonarchBase [@zhanMonarchBaseMonarchButterfly2013]. 

To determine if diapause is part of a generalized migratory syndrome in Australia, we also briefly re-analyzed the connection between reproductive status and wing morphology in our 2016 samples [@Freedman2018]. To determine if wing shape or size was correlated with reproductive status, we fit a pair of linear mixed effect models with either with wing shape or size as response variables, the number of yolked oocytes as a fixed effect, and maternal family as a random effect using the R package nlme [@Pinheiro2007].

# Results

## Phenotypic results and sequencing

Of the `r length(na.omit(sample.meta$quant_pheno))` total adult female monarch butterflies reared in both these studies and by Freedman et al. [-@Freedman2018], `r sum(sample.meta$quant_pheno == 2, na.rm = TRUE)`, `r sum(sample.meta$quant_pheno == 1, na.rm = TRUE)`, and `r sum(sample.meta$quant_pheno == 0, na.rm = TRUE)` were classified as reproductively fully mature, partially mature, and immature, respectively. We sequenced a total of `r prettyNum(total_reads, big.mark = ",")` reads across all individuals, `r round(percent_mapped, 3)*100`% of which mapped to the monarch reference genome and `r round(percent_retained, 3)*100`% of which were maintained after filtering. From this, we called genotypes and likelihoods for 179,735 and `r prettyNum(nrow(adat_quant),big.mark=",",scientific=FALSE)` SNP loci, respectively.

## Genetic basis of diapause
Maternal family strongly improved model fit (ANOVA, $p < 0.001$, see Figure 2b). From a total of `r prettyNum(nrow(adat_quant),big.mark=",",scientific=FALSE)`SNPs used in the association test, we identified three genomic regions which were significantly associated with reproductive development after false discovery rate correction on chromosomes 17, 19, and 29 (Figure 2d). A total of 26 genes were within 50kb of these regions (Table S1), one of which (LOC116776761/DPOGS204613)  was located directly under the significantly associated SNPs on chromosome 29 (minimum p-value = `r round(min(adat_quant$fdr), 3)` (Figure 3). No annotated genes were directly under the peaks of association on chromosomes 17 and 19 (Figure 3).

We found that wing length was significantly negatively correlated with the number of yolked oocytes in Freedman et al.'s [-@Freedman2018] earlier study ($p$ = `r round(summary(mod_AsRa)$tTable[2,5], 3)`) after accounting for maternal family. Wing shape was also negatively correlated with the number of yolked oocytes, but not significantly so after accounting for maternal family ($p$ = `r round(summary(mod_Area)$tTable[2,5], 3)`).

# Discussion

<!-- Outline: -->
<!--   The gene we found and why it makes sense -->
<!--   The gene we found in the context of migratory genes elsewhere in monarchs -->
<!--       This means that this is likely a novel mutation, not conserved variation -->
<!--   What this means for the persistence of migration in monarchs -->
<!--   What this means for the persistence and adaptability of migration in general. -->

## Evolution of diapause control in Australian monarchs

We found that a region on Chromosome 29 containing the gene DPOGS204613 (homologous to the gene *Karst* in *Drosophila* and hereafter referred to as such) was strongly associated with diapause initiation in monarch butterflies in Queensland, Australia (Figure 3c). According to the most recent annotation [@ranzNovoTranscriptionalAtlas2021], *Karst* codes for a spectrin beta chain protein involved in actin filament binding. *Karst* has not been previously identified as associated with migratory life history variation in North American monarchs. Specifically, *Karst* was not among the 536 genes located in regions of the genome found to be significantly associated with contrasts between migration and residency in North and South American, Pacific, and European populations by Zhan et al. [-@Zhan2014]. 

The involvement of *Karst* in migratory diapause initiation, however, is not surprising. Juvenile hormone mediated actin contraction is known to be involved in oocyte formation [@luoJuvenileHormoneSignaling2021], and mutations in *Karst* are known to cause serious issues during oogenesis in *Drosophila* [@zarnescuApicalSpectrinEssential1999]. The involvement of *Karst* in oocyte development is likely conserved in Lepidoptera: a predicted homolog of *Karst*, BGIBMGA012171, has been identified as an ovary-specific expressed protein in silkworms (*Bombyx mori*) [@chenProteomeAnalysisSilkworm2013a]. More directly, while it does not overlap with any of genes identified as involved in photoperiodic responses by Iiams et al. [-@iiamsPhotoperiodicClockRegulation2019], *Karst* has been observed to be differentially expressed during diapause termination in western North American monarchs [@GreenII2019].

There are at least two distinct explanations novel involvement of *Karst* in diapause initiation in Australian monarchs: 1) the diapause-associated *Karst* variant we observed is either present in North America but was not previously recognized as associated with migration or it 2) is the result of a new mutation which occurred in Australian monarchs. The former case would imply that this allele was either maintained as monarchs passed through repeated population bottlenecks during their Pacific expansion or was secondarily re-introduced by immigrant individuals from North America (which would have likely been anthropogenic in origin).

It seems unlikely that a secondary re-introduction of migratory monarch butterflies from North America could explain the re-emergence of migratory behavior in Australia, since there would be no reason for known migratory associated genes to have not also made the jump back across the Pacific. Ideally, it would be possible to delineate between the other two options (maintenance of variance at *Karst* during the Pacific crossing or a new mutation) by comparing the variance in *Karst* in Australia to that elsewhere, but that is not currently possible given that our data is from reduced-representation sequencing and we cannot therefore be confident that we have sequenced the causal loci. Further work that tests for a similar association between *Karst* genotypes and diapause initiation in North American monarchs would help to resolve whether Australian monarchs carry novel or ancestral migration-associated genetic variation at the *Karst* locus. 

Note that it is unlikely that *Karst* alone controls diapause induction under declining daylength in Australian monarchs. We detected two other strong associations with diapause, but it is not clear to which genes these outliers correspond given the lack of resolution in this study (Figure 3a-b). There are a few potential reasons for this: first, it is possible that the causal variants are located in transcription factor binding cites or other promoter regions which are not well annotated. Alternatively, we may not have sequence data for the actual causal variants with which our loci are in partial linkage disequilibrium. Higher resolution sequencing could again help clarify the causal genes for these regions.

Additionally, while our study features many *individuals*, it features only 32 maternal families and thus far fewer independent samples. Our power is therefore limited, and it is entirely likely that we failed to detect many causal loci for diapause onset.  Additional studies with larger sample sizes are therefore still needed to better understand the mechanisms underlying diapause induction in Australian monarch butterflies.

Regardless, it is interesting that *Karst* had not been previously noted as associated with migration in monarch butterflies. In their 2014 study, Zhan et al. generally focused their discussion of migratory genetics on Collagen IV $\alpha$-1, one of three genes they found that were strong outliers for association with migration. This gene is involved in muscle functioning in insects [@Schnorrer2010], and Zhan et al. hypothesized that divergence at this gene was driven by selection for increased wing length and muscle efficiency to facilitate long-distance migration. While they mentioned that the remaining significantly associated genes were enriched for the "morphogenesis, neurogenesis, and extracellular matrix/basement membrane" functional terms, they were not otherwise discussed (*Karst* does not have any of these functional terms). It is possible that *Karst* also contributes to diapause and migration in butterflies outside of Australia, but that the quantitative effect of other genes (a product of the effect sizes of the migratory alleles and their frequencies) dwarfs that of *Karst*, thus concealing the relative impact of the latter gene.

We did not identify loci near the Collagen IV $\alpha$-1 gene as associated with diapause, despite the fact that we did sequence several nearby loci. This is surprising, since although monarchs in Australia (like other non-migratory populations), have substantially smaller and shorter wings than their migratory counterparts [@Freedman2020], we also found from re-analyzing data from Freedman et al.'s (2018) study that wing length is significantly correlated with reproductive development in Australian monarchs. This implies that wing morphology is still a part of the correlated migratory syndrome in Australian monarchs, and so we might expect to see some potential causal genetic elements for wing morphology to also be associated with diapause. 

However, it does make some sense that variance in Collagen IV $\alpha$-1 would not be key to migratory life history variation in populations that are *newly* migratory. Variance in genes that code for traits such as muscle strength or endurance, such as Collagen IV $\alpha$-1, could make already migratory individuals more fit, but would not itself *cause* or *enable* migratory behavior. *Karst* and other genes that code for reproductive diapause may therefore be involved earlier in the evolution of migration, since individuals that do not delay their reproductive investment have much shorter life-spans and often cannot complete their full migration [@Herman2001]. Genes that trigger migratory behavior or control orientation, navigation, and directed flight would also fall into this category to varying extents. That we found *Karst* specifically to be associated with diapause in Australia is therefore not surprising for a newly migratory population [@merlinGeneticsEpigeneticsAnimal2019; @greenMonarchButterflyMigration2021]. 

## Persistence of a migratory life history in Australian monarchs

Our work suggests that even after ceasing migration for hundreds of generations, monarchs may maintain genetic variation that underlies migratory plasticity. Despite the loss of allelic variance during the successive bottlenecks, monarch butterflies either maintained the genetic variation needed to migrate or secondarily re-acquired it via mutation. This gives us some hope that the contemporary loss of migration that we have observed in many migratory species may be reversible over relatively short evolutionary timescales. At the very least, we can be assured that for monarchs in particular, migratory populations in Australia constitute a reservoir of migratory alleles that could potentially be tapped for North America if needed.

While monarchs may be able to quickly recover from the loss of migratory behavior, it is unlikely that this is the case for most other migratory species of conservation concern, particularly for vertebrates. For example, if we assume the latter hypothesis (migratory variance in Australian monarchs is the result of a new mutation), it is clear that monarchs have had ample evolutionary opportunity to re-acquire migratory mutations: monarchs have a generation time of approximately seven generations per year if continuously breeding, which means that monarchs in Australia went through roughly ~ `r 7*50` generations in the 50 or so years since they were first reported on the continent, and although they likely experienced a strong bottleneck initially, they were probably at a relatively large effective size for most of that time. Since the rate at which new mutations appear in a population is proportional to both effective populations size and generation time, monarchs have had a large opportunity space for the generation of new migratory alleles. 

Alternatively, the maintenance of ancestral, North American migratory variance in Australian monarchs would only have been possible because selection was not actively acting against that variance. This is actually quite likely: monarchs in the Pacific are not exposed to substantial seasonal changes in day length or average temperature, and thus do not receive the primary cues thought to be associated with migration initiation. Environmentally triggered, phenotypically plastic migratory variation, therefore, is probably nearly neutral in the Pacific. In contrast, individually fixed migratory variance, such as wing morphology, was likely selected against across the Pacific, thus driving the observed, repeated decreases in wing size and length observed in newly non-migratory monarch populations [@Freedman2020]. Environmentally triggered migratory-associated genetic variation is therefore more likely to be maintained in non-migratory populations than that underlying phenotypically fixed traits. 

# Conclusion
This study suggests that reproductive diapause in Australian monarch butterflies is partially influenced by a novel genetic mechanism via the *Karst* gene, and that the previously identified variable migratory genomic regions were not associated with diapause in this populations. Research to determine if the migratory variant of this gene is the result of ancestral variation maintained from North America or a new mutation is needed. Our work is also consistent with the hypothesis that genetic variance which enables migration, rather than that which bolsters the fitness of already migratory individuals, is more likely to be observed in recently evolved migratory populations.

# Acknowledgements and Funding Sources
We would like to thank Dr. Shawan Chowdhury for his assistance with monarch feeding and maintenance during the rearing process and Dr. Sean O’Rourke for assistance during library preparation and sequencing. Travel and sequencing funding was provided by the Jastro-Shields Graduate Research Award and the UC Davis Department of Animal Science. MF was funded by NSF Postdoctoral Fellowship Award 2010658.


# References
<div id="refs"></div>

# Data Availability and Benefits-Sharing
Data Availability: The raw sequencing data produced for this study are available from the NCBI under accession number PRJNA984345. The filtered genotypes and the scripts used to produce both them and this paper are available at https://github.com/hemstrow/aus_monarchs. The exact script used to produce this paper from the genotypic data, including all non-map figures, is available at https://github.com/hemstrow/aus_monarchs/blob/master/paper/paper_draft.Rmd.

Benefit-Sharing: This paper represents the continuation of a collaboration between researchers in the United States and Australia. All major contributors are listed as co-authors. This paper also extends our knowledge of critical phenotypic variance in a newly-listed endangered species. All of the data produced in this paper are shared with the public at the repositories above.

# Author Contributions
WH and MM designed the study with input from MF and MZ. WH collected, raised, and phenotyped individuals. MZ provided, lab space, equipment, and expertise for each of those steps. WH analyzed the data and wrote the manuscript. MF, MZ, and MM provided feedback on the analysis and manuscript.


# Tables and Figures

```{r Table1,echo=FALSE}
Tab1 <- table(sample.meta[,c("year", "quant_pheno")])
Tab1 <- addmargins(Tab1)
Tab1 <- reshape2::dcast(as.data.frame(Tab1), year ~ quant_pheno, value.var = "Freq")
Tab1$year <- as.character(Tab1$year)
Tab1[Tab1 == "Sum"] <- "Total"
colnames(Tab1) <- c("Year", "# Immature", "# Partially Mature", "# Mature", "Total")
# Tab1 <- gt::gt(Tab1, caption = c("Table 1: Number of individuals in diapuase or that are reproductive for each year.")) %>%
#   gt::cols_label(year = "Year", M = gt::html("N<sub>subscript</sub>"), R = gt::html("N<sub>Reproductive</sub>"), Sum = gt::html("N<sub>Total</sub>"))


# kableExtra::kable(Tab1, format = "latex", booktabs = T, linesep ="\\addlinespace", caption = c("Table 1: Number of individuals in diapuase or that are reproductive for each year."), col.names = c("Year", "Diapause", "Reproductive", "Total")) %>% kableExtra::kable_styling()

# kableExtra::kable(Tab1,
#              caption = "Table 1: Contingency Table for the number of diapaused and reproductive individuals for each year.",
#              col.names = c("Year", "# Diapause", "# Reproductive", "# Total"), escape = FALSE) %>% kableExtra::kable_styling()


flextable::regulartable(Tab1, cwidth = 1.2) %>% flextable::set_caption(., "Total count of immature, partially mature, and mature female monarch butterflies reared and sequenced in each sampling year.")
```

Figure 1: Approximate ancestral North American and introduced migratory and non-migratory ranges with introduction dates for monarch butterflies in the Pacific, with sampling location noted in blue. Photo courtesy of William Hemstrom.

Figure 2: Genome-wide association results for diapause in Australian monarch butterflies. A: Examples of reproductive development for female adult monarchs rates as reproductively mature, partially mature, and immature. B: Average reproductive development scores for maternal families. Average scores were significantly more skewed than expected by chance (orange line). C: Quantile-quantile plot for association results showing a few strong outliers but otherwise conformance to the expected distribution of p-values. D: Manhattan plot showing the strength of association with reproductive status genome-wide. Values above the blue line indicate significant associations after false discovery rate p-value adjustment.

Figure 3: Distribution of genome-wide association test p-values adjacent to each of the three outlier regions (top). Genes that are at least partially within 50kb of outlier SNPs are noted (bottom), with CDS regions highlighted in black. One specific gene (LOC133319386) had no match in the monarch official gene set, and so is instead listed with its RefSeq ID.